client/my-sites/plans-features-main/hooks/use-generate-action-hook.tsx

```diff
@@ -126,229 +133,360 @@ function useGenerateActionHook( {
 		 */
 		if ( isInSignup ) {
-			let text = translate( 'Get %(plan)s', {
-				args: {
-					plan: planTitle ?? '',
-				},
+			return getSignupAction( {
+				getActionCallback,
+				planSlug,
+				cartItemForPlan,
+				selectedStorageAddOn,
+				translate,
+				isLargeCurrency,
+				isStuck,
+				planTitle,
+				priceString,
+				isFreeTrialAction,
+				eligibleForFreeHostingTrial,
+				plansIntent,
 			} );
-			let postButtonText;
+		}
 
-			if ( isFreeTrialAction ) {
-				text = translate( 'Try for free' );
-			} else if ( isFreePlan( planSlug ) ) {
-				text = translate( 'Start with Free' );
-			} else if ( isStuck && ! isLargeCurrency ) {
-				/**
-				 * `isStuck` indicates the buttons are fixed/sticky in the grid, and we show the price alongside the plan name.
-				 */
-				text = translate( 'Get %(plan)s – %(priceString)s', {
-					args: {
-						plan: planTitle ?? '',
-						priceString: priceString ?? '',
-					},
-					comment:
-						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
-					// TODO: Revisit this type and why we have to force inference of string
-				} ) as string;
-			} else if ( isStuck && isLargeCurrency ) {
-				/**
-				 * `isStuck` indicates the buttons are fixed/sticky in the grid, and we show the price alongside the plan name.
-				 */
-				text = translate( 'Get %(plan)s {{span}}%(priceString)s{{/span}}', {
-					args: {
-						plan: planTitle ?? '',
-						priceString: priceString ?? '',
-					},
-					comment:
-						'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
-					components: {
-						span: <span className="plan-features-2023-grid__actions-signup-plan-text" />,
-					},
-				} );
-			}
+		/**
+		 * 4. Logged-In (Admin) Plans actions
+		 */
+		return getLoggedInPlansAction( {
+			getActionCallback,
+			planSlug,
+			cartItemForPlan,
+			selectedStorageAddOn,
+			translate,
+			isLargeCurrency,
+			isStuck,
+			planTitle,
+			priceString,
+			sitePlanSlug,
+			availableForPurchase,
+			domainFromHomeUpsellFlow,
+			canUserManageCurrentPlan,
+			isPlanExpired,
+			currentPlanBillingPeriod,
+			billingPeriod,
+		} );
+	};
 
-			if (
-				isBusinessPlan( planSlug ) &&
-				! eligibleForFreeHostingTrial &&
-				plansIntent === 'plans-new-hosted-site'
-			) {
-				postButtonText = translate( "You've already used your free trial! Thanks!" );
-			}
+	return useActionHook;
+}
 
-			return {
-				primary: {
-					callback: getActionCallback( { planSlug, cartItemForPlan, selectedStorageAddOn } ),
-					status: 'enabled',
-					text,
-				},
-				postButtonText,
-			};
-		}
+function getLaunchPageAction( {
+	getActionCallback,
+	planSlug,
+	cartItemForPlan,
+	selectedStorageAddOn,
+	translate,
+	isLargeCurrency,
+	isStuck,
+	planTitle,
+	priceString,
+}: {
+	getActionCallback: UseActionCallback;
+	planSlug: PlanSlug;
+	translate: LocalizeProps[ 'translate' ];
```

Comment:
@Automattic/i18n There are many other instances. It's certainly valid, but I'm wondering if that means we should export `typeof translate` from the packages directly, so we won't need to access nested type like this.